From 8817f4477402a1fd64502ab9a1d5680cd26f5e8f Mon Sep 17 00:00:00 2001
From: "yadi.hu@windriver.com" <yadi.hu@windriver.com>
Date: Wed, 8 Apr 2015 11:15:15 +0800
Subject: [PATCH] x86/byt: Sync DMA clock setting with YB BSP

commit 	1b275ee65e22fedc46ad82732498783bc15cf280 upstream

setup clock tree for PCI mode SPI, DMA and PWM host

Signed-off-by: Hu <yadi.hu@windriver.com>
---
 arch/x86/platform/byt/byt-board.c |   32 +++++++++++---------------------
 1 files changed, 11 insertions(+), 21 deletions(-)

diff --git a/arch/x86/platform/byt/byt-board.c b/arch/x86/platform/byt/byt-board.c
index 2bd265a..9550e4f 100644
--- a/arch/x86/platform/byt/byt-board.c
+++ b/arch/x86/platform/byt/byt-board.c
@@ -43,34 +43,23 @@ static int byt_spi_board_setup(void)
 
 static int byt_clk_setup(void)
 {
+
 	struct clk *clk;
 
-	/* Make sure the root clk required by the LPSS driver is registered */
-	clk = clk_get(NULL, "lpss_clk");
-	if (IS_ERR(clk)) {
-		clk = clk_register_fixed_rate(NULL, "lpss_clk", NULL, CLK_IS_ROOT,
-							100000000);
-	}
+	/* Make clock tree required by the SPI/DMA/PWM driver */
+	clk = clk_register_fixed_rate(NULL, "lpss_clk", NULL, CLK_IS_ROOT,
+								100000000);
+	if (IS_ERR(clk))
+		return PTR_ERR(clk);
+
+	clk_register_clkdev(clk, "hclk", "0000:00:1e.0");
 
-	/* 
-	 * to check has the spi_clk been registered by the ACPI mode,
-	 * if yes, skip it, otherwise, register those clks.
-	*/
-	clk = clk_get(NULL, "spi_clk");
-	if (IS_ERR(clk)) {
-		clk = clk_register_fixed_rate(NULL, "spi_clk", "lpss_clk", 0, 50000000);
-		if (IS_ERR(clk))
-			return PTR_ERR(clk);
-	} else
+	clk = clk_register_fixed_rate(NULL, "spi_clk", "lpss_clk", 0, 50000000);
+	if (IS_ERR(clk))
 		return PTR_ERR(clk);
 
 	clk_register_clkdev(clk, NULL, "0000:00:1e.5");
 
-	/* register the clock for DW DMA engines */
-	clk_register_clkdev(clk, "hclk", "0000:00:18.0");
-	clk_register_clkdev(clk, "hclk", "0000:00:1e.0");
-
-	/* Make clock tree required by the PWM driver */
 	clk = clk_register_fixed_rate(NULL, "pwm_clk", "lpss_clk", 0, 25000000);
 	if (IS_ERR(clk))
 		return PTR_ERR(clk);
@@ -79,6 +68,7 @@ static int byt_clk_setup(void)
 	clk_register_clkdev(clk, NULL, "0000:00:1e.2");
 
 	return 0;
+
 }
 
 static int __init byt_board_init(void)
-- 
1.7.5.4

